CAT-2151 Fixed crash when whatsapp not installed

chandruscm opened the pull request:
Social sharing via the WhatsApp button on details page in the webview crashes PocketCode when WhatsApp is not installed on the phone.

*Added code to check WhatsApp installation before starting the share intent.

catrobatjenkins commented:
Can a catrobat member please verify this patch? If it doesn't look evil comment with "ok to test" to start the testrun.

cheidenreich commented:
please fix the 5 introduced warnings, otherwise looking good

chandruscm commented:
@cheidenreich Could you please paste those warnings? Are they arising from this commit? I can't find them.

catrobatjenkins commented:
Can a catrobat member please verify this patch? If it doesn't look evil comment with "ok to test" to start the testrun.

cheidenreich commented:
@chandruscm hey,
sorry for the delay, currently something is being configured on our test server, i'll get back to you as soon as it's done ;)

chandruscm commented:
@cheidenreich Hey it's fine, please take your time :) I also have a noob question pertaining to JIRA. How do I make my commits appear in a JIRA ticket, when I make a pull request, JIRA automatically picks it up. What about commits? I make them in my local branch and then?

cheidenreich commented on catroid/src/main/java/org/catrobat/catroid/ui/WebViewActivity.java:
```diff
@@ -71,6 +72,7 @@
 	private String callingActivity;
 	private ProgressDialog progressDialog;
 	private Intent resultIntent = new Intent();
+	private final String PACKAGE_NAME_WHATSAPP = "com.whatsapp";
```
ok i just checked the test run, the warning comes from line 75:

```
Member name not following naming convention - Name 'PACKAGE_NAME_WHATSAPP' must match pattern '^(([a-z]{2})|([x-z][A-Z]))[a-zA-Z0-9]*$'.

Validates identifiers for non-static fields.
```

but instead of changing the name I'd rather make the variable static, since it doesn't need to be instanced ;)

cheidenreich commented:
@chandruscm I did another test run with your PR, seems like there is only 1 warning (I've commented the exact warning in line in the diff view).

Regarding JIRA: It is linked to the official Catrobat repository, so if you create a PR, it registeres this as it is made on this repository. Commits from forked repositories appear only after they are merged into this one.

Hope that answers your question ;)

BR

chandruscm commented:
@cheidenreich I've changed it to static and moved it below the rest of the private static variables.

Regarding JIRA, it makes sense now. Thanks a lot :)

cheidenreich commented:
@chandruscm thanks for the fast change, unfortunately there seems to be a conflict with the current develop branch now. could you please rebase your branch?

sorry for the inconvenience.

BR

catrobatjenkins commented:
Can a catrobat member please verify this patch? If it doesn't look evil comment with "ok to test" to start the testrun.

chandruscm commented:
@cheidenreich Please review, I've rebased the branch with 2 changes.

1. Squashed my previous 2 commits.
2. Moved the line `private ProgressDialog webViewLoadingDialog;` below `private ProgressDialog progressDialog;` for consistency.

cheidenreich commented:
@chandruscm looking good now, thanks for the patience ;)

cheidenreich commented:
@chandruscm 
just a heads up for the future: if you create a new branch on your repository when working on a ticket, it will help you keeping your own repository clean, especially if you're working on multiple tickets ;)

chandruscm commented:
@cheidenreich Learned so much from you, thanks a lot :) Look out for my next PR :D
